### ISSUE TITLE
Cannot exit ipython with running qcodes.Monitor()

### ISSUE THREAD

**Issue by michaelc12qe** [QCoDeS/Qcodes#5561]
### Steps to reproduce
iPython version 8.10, qcodes version 0.42, Windows 10 Pro 22H2

1. open ipython
2. from qcodes import Monitor
monitor = Monitor()
3. exit ipython

It hangs instead of closing iPython. The terminal must be terminated.

**Comment by jenshnielsen** [comment]
I can reproduce this. Not sure what is going on but perhaps related to https://github.com/QCoDeS/Qcodes/issues/3814 

It would be nice to investigate if changing the monitor thread to be a deamon thread would also resolve this issue

**Comment by jenshnielsen** [comment]
Note that until this is resolved it can be mitigated by explicitly calling join on the monitor thread

```
❯ ipython
Python 3.11.5 | packaged by Anaconda, Inc. | (main, Sep 11 2023, 13:26:23) [MSC v.1916 64 bit (AMD64)]
Type 'copyright', 'credits' or 'license' for more information
IPython 8.18.1 -- An enhanced Interactive Python. Type '?' for help.

In [1]: from qcodes import Monitor
In [2]: m = Monitor()
In [3]: m.join()
In [4]: exit

~ took 29s
❯
```

**Comment by michaelc12qe** [comment]
I've tested turning the monitor thread into a daemon thread as you suggested, by passing the keyword argument `daemon=True` into the `Monitor` initialization. It appears to resolve this issue. I've also checked that the behaviour of the monitor remains unchanged by add QDAC2 voltage channels to the monitor, and it seems fine.

I don't know if there's any more testing required, but happy to raise a PR for this if you like.
